glib-macros/tests/properties.rs

```diff
@@ -34,9 +34,13 @@ fn props() {
             pub struct Foo {
                 #[prop(get, set)]
                 bar: Mutex<String>,
+                // The following property doesn't store any data. The value of the property is calculated
+                // when the value is accessed.
                 #[prop(get = Self::hello_world)]
                 _buzz: PhantomData<String>,
-                #[prop(get, set = Self::set_fizz, name = "fizz")]
+                #[prop(get, set = Self::set_fizz, name = "fizz", nick = "fizz-nick",
+                    blurb = "short description stored in the GLib type system"
```

Comment:
Wouldn't it be nice if we can also use the native Rust way of setting the description of a field?

Reply:
So we can try something like 

```rs
#[derive(Properties, Default)]
pub struct Foo {
 /// The description stored in the Glib type system will use native Rustdoc comments
 #[property(get, set)
 fizz: RefCell<String>,
  
}
```

Reply:
Also, what do you think of making the syntax uses keywords rather than attributes? So we can try something like this:

```rs
#[derive(Properties, Default)]
struct Foo {
// will enable get and set 
rw fizzbar: RefCell<String>;

// will disable set
ro fizz: RefCell<String>
}
```

Reply:
OOohh! I have an idea. 😄

What if we combine this macro with `glib::wrapper!`?


```rs
glib::wrapper! {
pub struct FizzbarBox 
@extends gtk::Widget, gtk::Box,
@implements gtk::Accessible, gtk::Buildable, gtk::ConstraintTarget, gtk::Orientable
{
   rw fizzbar: RefCell<String>,
   ro fizz: RefCell<String>,
   wo fizz_phantom: PhantomData<String>
}

}
```

Reply:
Nah, attributes are much better than keywords because they are easier for tooling to understand. They are also easier for newcomers compared to completely new syntax.

Reply:
Okay, then.

Reply:
>  The description stored in the Glib type system will use native Rustdoc comments

I'm not sure I agree... glib's "blurb" should be a _short_ description, while Rustdoc comments don't have any length constraints. @jf2048 what do you think?

Reply:
> What if we combine this macro with `glib::wrapper!`?

A future `class` macro will combine everything: subclass, wrapper, properties, signals... See #540.

Reply:
> A future `class` macro will combine everything: subclass, wrapper, properties, signals... See #540.

Ohh. Thanks so much!

Reply:
> 
> I'm not sure I agree... glib's "blurb" should be a _short_ description, while Rustdoc comments don't have any length constraints. @jf2048 what do you think?
> 

I think we can try to get the first line of the comment?

```
The avatar url of the user.

Example:
https://example.com/avatar?id=user_id
```

The Glib blurb will be just `The avatar url of the user.` without the rest of the lines.

Reply:
Upstream has disincentivized the use of blurb and nick: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4717

These don't really have a use case, only the rustdoc comments will matter.

Reply:
That's from a GTK point of view and is otherwise false. For GStreamer we even depend on these for properties provided by plugins.

Reply:
I am not aware of any uses for these besides gstreamer.

Reply:
I'll leave this as is now => no auto generated blurb

Reply:
@jf2048 GStreamer is the biggest user of the dynamic features of GObject, so that's not very surprising :)